tools/3d_tensor_feature_dimension_reduction/3d_tensor_feature_dimension_reduction.xml

```diff
@@ -0,0 +1,38 @@
+<tool id="ip_3d_tensor_feature_dimension_reduction" name="Dimensionality Reduction" version="0.0.1" profile="20.05"> 
+    <description>for features of 3D tensor data using UMAP</description>
+    <requirements>
+        <requirement type="package" version="1.20.2">numpy</requirement>
+        <requirement type="package" version="3.6.0">h5py</requirement>
+        <requirement type="package" version="2020.10.1">tifffile</requirement>
+        <requirement type="package" version="0.5.2">umap-learn</requirement>
+    </requirements>
+    <command detect_errors="aggressive"><![CDATA[
+        ln -s '$fn_in' ./input.h5 &&
+        python '$__tool_directory__/3d_tensor_feature_dimension_reduction.py'
+        ./input.h5
+        '$nCh'
+        ./output.tif
+    ]]>
+    </command>
+    <inputs>
+        <param name="fn_in" type="data" format="hdf5" label="3D tensor data (rows x cols x features)" />
+        <param name="nCh" type="integer" value="5" optional="true" min="1" max="10" label="Reduced dimensions of features" />
+    </inputs>
+    <outputs>
+        <data format="tiff" name="fn_out" from_work_dir="output.tif" />
+    </outputs>
+    <tests>
+        <test>
+            <param name="fn_in" value="tensor.h5"/>
+            <param name="nCh" value="3"/>
+            <output name="fn_out" value="tensor_r.tif" ftype="tiff" />
```

Comment:
by default galaxy will diff your files, this is not possible with binary data. You can use for example this here: https://docs.galaxyproject.org/en/latest/dev/schema.html#tool-tests-test-output-assert-contents or any otherway to test that your tool is producing what you expect :)

Reply:
Now I check the output file size